lego/apps/events/serializers/events.py

```diff
@@ -269,6 +281,15 @@ def validate(self, data):
     def create(self, validated_data):
         pools = validated_data.pop("pools", [])
         is_abakom_only = validated_data.pop("is_abakom_only", False)
+        event_status_type = validated_data.get("event_status_type", constants.NORMAL)
+        if event_status_type == constants.TBA:
+            pools = []
+            validated_data["location"] = "TBA"
```

Comment:
Shouldn't you be able to set a location if you have one, even if the event itself is TBA?

Reply:
I don't think so. This will open more cases. TBA events that have a location, and TBA events that don't have a location. I think we should make it clear and simple, this is an event that is to be announced.

Reply:
I honestly agree, it's easier for us just to set the location as TBA, and let them rather set the location once it's no longer TBA.

I'll update the PR to reflect this tomorrow.

Reply:
90% of all TBA events have 'TBA' as location anyhow.

Reply:
Isn't the point of this to not have to use the location to determine what type of event, so is it really another case to have a location?

Reply:
I don't see a reason to restrict the location on events of type TBA. What function does it have? We can style the TBA event in frontend, and you cannot create pools unless the event is not changed to normal mode anyways.

Reply:
It's just less to fill out when creating a `TBA` event. Now they need to look up if they maybe have a location and fill that since it's a required field or write TBA themselves.

And when switching to a normal event I want the `location` field to be empty, so they don't forget to switch that from `TBA/random`.